### PR TITLE
Improve docs consistency for Overlay, PageHead, Pagination, Popover, Portal, ProgressBar components

### PR DESCRIPTION
**Pull request by rezrah**
Part of #1701

Added / Updated docs for:

- [x] `Overlay` ([see it here](https://primer-react-qkijen9i5-primer.vercel.app/react/Overlay)) 
- [x] `PageHead`  ([see it here](https://primer-react-qkijen9i5-primer.vercel.app/react/Pagehead)) 
- [x] `Pagination` ([see it here](https://primer-react-qkijen9i5-primer.vercel.app/react/Pagination)) 
- [x] `Popover` ([see it here](https://primer-react-qkijen9i5-primer.vercel.app/react/Popover)) 
- [x] `Portal` ([see it here](https://primer-react-qkijen9i5-primer.vercel.app/react/Portal)) 
- [x] `ProgressBar` ([see it here](https://primer-react-qkijen9i5-primer.vercel.app/react/ProgressBar)) 

### Merge checklist

- [ ] Added/updated tests
- [x] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

File: docs/src/@primer/gatsby-theme-doctocat/nav.yml [hunk 1]
```diff
@@ -85,8 +85,8 @@
       url: /LabelGroup
     - title: Link
       url: /Link
-    # - title: Overlay
-    #   url: /Overlay
+    - title: Overlay
+      url: /Overlay
```
Comment (rezrah) [inline docs/src/@primer/gatsby-theme-doctocat/nav.yml:89]: Unclear why this was hidden. Does anyone know?

File: docs/content/Pagination.md [hunk 2]
```diff
@@ -45,7 +49,7 @@ By default, clicking a link in the pagination component will cause the browser t
 </State>
 ```
 
-## Customizing link URLs
+### Customizing link URLs
```
Comment (siddharthkp) [inline docs/content/Pagination.md:52]: ❤️

**Review by siddharthkp — APPROVED** [review]
Love to see it! ❤️

File: docs/content/Overlay.mdx [hunk 3]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+    }
+  />
+  <PropsTableRow
+    name="width"
+    type={
+      <>
+        <InlineCode>small</InlineCode>, <InlineCode>medium</InlineCode>, <InlineCode>large</InlineCode>,{' '}
+        <InlineCode>xlarge</InlineCode>, <InlineCode>xxlarge</InlineCode>, <InlineCode>auto</InlineCode>
+      </>
+    }
```
Comment (colebemis) [inline docs/content/Overlay.mdx:152]:
```suggestion
    type="'small' | 'medium' | 'large' | 'xlarge' | 'xxlarge' | 'auto'"
```

File: docs/content/Overlay.mdx [hunk 4]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+    defaultValue="auto"
```
Comment (colebemis) [inline docs/content/Overlay.mdx:153]:
```suggestion
    defaultValue="'auto'"
```

File: docs/content/Overlay.mdx [hunk 5]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+    }
+  />
+  <PropsTableRow
+    name="height"
+    type={
+      <>
+        <InlineCode>xsmall</InlineCode>, <InlineCode>small</InlineCode>, <InlineCode>medium</InlineCode>,{' '}
+        <InlineCode>large</InlineCode>, <InlineCode>xlarge</InlineCode>, <InlineCode>auto</InlineCode>
+      </>
+    }
```
Comment (colebemis) [inline docs/content/Overlay.mdx:172]:
```suggestion
    type="'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | 'auto'"
```

File: docs/content/Overlay.mdx [hunk 6]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+    defaultValue="auto"
```
Comment (colebemis) [inline docs/content/Overlay.mdx:173]:
```suggestion
    defaultValue="'auto'"
```

File: docs/content/Overlay.mdx [hunk 7]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+      </>
+    }
+  />
+  <PropsTableRow
+    name="visibility"
+    type={
+      <>
+        <InlineCode>visible</InlineCode>, <InlineCode>hidden</InlineCode>
+      </>
+    }
```
Comment (colebemis) [inline docs/content/Overlay.mdx:191]:
```suggestion
    type="'visible' | 'hidden'"
```

File: docs/content/Overlay.mdx [hunk 8]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+    defaultValue="visible"
```
Comment (colebemis) [inline docs/content/Overlay.mdx:192]:
```suggestion
    defaultValue="'visible'"
```

File: docs/content/Overlay.mdx [hunk 9]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+    name="anchorSide"
+    type={
+      <>
+        <InlineCode>inside-top</InlineCode>, <InlineCode>inside-bottom</InlineCode>,{' '}
+        <InlineCode>inside-left</InlineCode>, <InlineCode>inside-right</InlineCode>,{' '}
+        <InlineCode>inside-center</InlineCode>, <InlineCode>outside-top</InlineCode>,{' '}
+        <InlineCode>outside-bottom</InlineCode>, <InlineCode>outside-left</InlineCode>,{' '}
+        <InlineCode>outside-right</InlineCode>
+      </>
+    }
```
Comment (colebemis) [inline docs/content/Overlay.mdx:209]:
```suggestion
    type="'inside-top' | 'inside-bottom' | 'inside-left' | 'inside-right' | 'inside-center' | 'outside-top' | 'outside-bottom' | 'outside-left' | 'outside-right'"
```

File: docs/content/Overlay.mdx [hunk 10]
```diff
@@ -64,24 +56,212 @@ const Demo = () => {
...
+      <>
+        Horizontal position of the overlay, relative to its closest positioned ancestor (often its{' '}
+        <InlineCode>Portal</InlineCode>).
+      </>
+    }
+  />
+  <PropsTableRow
+    name="portalContainerName"
+    type="string"
+    defaultValue={0}
```
Comment (colebemis) [inline docs/content/Overlay.mdx:239]:
```suggestion
```

File: docs/content/Overlay.mdx (hunk 5 above)
Comment (rezrah) [reply docs/content/Overlay.mdx]: @colebemis. Thanks. I've rolled back most of these changes to match your suggestions.

I do feel however, that it's less readable this way, what do you think? 

E.g.

Before:
Screenshot 2022-01-18 at 20 38 03

After:
Screenshot 2022-01-18 at 20 43 32

Comment (colebemis) [reply docs/content/Overlay.mdx]: Agreed. Looks like we use newlines in `AnchoredOverlay` docs to solve this issue: https://primer.style/react/AnchoredOverlay#props

We could do that here too

Comment (colebemis) [reply docs/content/Overlay.mdx]: CleanShot 2022-01-18 at 13 51 50@2x

```
  <PropsTableRow
    name="side"
    type={`| 'inside-top' 
| 'inside-bottom' 
| 'inside-left' 
| 'inside-right' 
| 'inside-center' 
| 'outside-top' 
| 'outside-bottom' 
| 'outside-left' 
| 'outside-right'`}
    defaultValue="'outside-bottom'"
  />
```

Comment (rezrah) [reply docs/content/Overlay.mdx]: Much better. Thanks @colebemis
Screenshot 2022-01-19 at 13 40 53

File: docs/content/Popover.md [hunk 11]
```diff
@@ -107,25 +111,75 @@ function CaretSelector(props) {
...
+    description="Sets the underlying HTML tag for the component"
+  />
+  <PropsTableRow
+    name="caret"
+    defaultValue="top"
+    type={
+      <>
+        <InlineCode>top</InlineCode>, <InlineCode>bottom</InlineCode>, <InlineCode>left</InlineCode>, <InlineCode>right</InlineCode>, <InlineCode>bottom-left</InlineCode>, <InlineCode>bottom-right</InlineCode>, <InlineCode>top-left</InlineCode>, <InlineCode>top-right</InlineCode>, <InlineCode>left-bottom</InlineCode>, <InlineCode>left-top</InlineCode>, <InlineCode>right-bottom</InlineCode>, or <InlineCode>right-top</InlineCode>
+      </>
+    }
```
Comment (colebemis) [inline docs/content/Popover.md:132]:
```suggestion
    defaultValue="'top'"
    type={`| 'top'
| 'bottom'
| 'left'
| 'right'
| 'bottom-left'
| 'bottom-right'
| 'top-left'
| 'top-right'
| 'left-bottom'
| 'left-top'
| 'right-bottom'
| 'right-top'` }
```

File: docs/content/Popover.md [hunk 12]
```diff
@@ -107,25 +111,75 @@ function CaretSelector(props) {
...
-#### Caret Positions
-
-The `caret` prop can be one of the following values: `top`, `bottom`, `left`, `right`, `bottom-left`, `bottom-right`, `top-left`, `top-right`, `left-bottom`, `left-top`, `right-bottom`, or `right-top`.
+<PropsTable>
+  <PropsTableRow
+    name="as"
+    defaultValue="div"
+    type="string"
+    description="Sets the underlying HTML tag for the component"
+  />
```
Comment (colebemis) [inline docs/content/Popover.md:124]:
```suggestion
  <PropsTableAsRow
    defaultElementType="div"
  />
```

File: docs/content/Pagehead.md [hunk 13]
```diff
@@ -1,20 +1,52 @@
...
+
+### Pagehead
+
+<PropsTable>
+  <PropsTableRow
+    name="as"
+    defaultValue="div"
+    type="string"
+    description="Sets the underlying HTML tag for the component"
+  />
```
Comment (colebemis) [inline docs/content/Pagehead.md:29]:
```suggestion
  <PropsTableAsRow
    defaultElementType="div"
  />
```

File: docs/content/Popover.md [hunk 14]
```diff
@@ -107,25 +111,75 @@ function CaretSelector(props) {
...
 
 ### Popover.Content
 
+<PropsTable>
+  <PropsTableRow
+    name="as"
+    defaultValue="div"
+    type="string"
+    description="Sets the underlying HTML tag for the component"
+  />
```
Comment (colebemis) [inline docs/content/Popover.md:158]:
```suggestion
  <PropsTableAsRow
    defaultElementType="div"
  />
```

File: docs/content/Popover.md [hunk 15]
```diff
@@ -107,25 +111,75 @@ function CaretSelector(props) {
...
+  <PropsTableSxRow />
+</PropsTable>
+
 | Name | Type              | Default | Description                          |
 | :--- | :---------------- | :-----: | :----------------------------------- |
 | as   | String            |  'div'  | Sets the HTML tag for the component. |
 | sx   | SystemStyleObject |   {}    | Style to be applied to the component |
```
Comment (colebemis) [inline docs/content/Popover.md:165]:
```suggestion
```

File: docs/content/ProgressBar.mdx [hunk 16]
```diff
@@ -1,16 +1,22 @@
...
 status: Alpha
 source: https://github.com/primer/react/blob/main/src/ProgressBar.tsx
 ---
 
-## Default example
+```js
+import {ProgressBar} from '@primer/react'
+```
+
+## Example
```
Comment (colebemis) [inline docs/content/ProgressBar.mdx:13]:
```suggestion
## Examples

### Default

```

File: docs/content/ProgressBar.mdx [hunk 17]
```diff
@@ -20,12 +26,43 @@ If you'd like to use ProgressBar inline, pass the `inline` boolean prop & **be s
...
+
+<PropsTable>
+  <PropsTableRow name="progress" type="number" description="Used to set the size of the green bar" />
+  <PropsTableRow
+    name="barSize"
+    type={
+      <>
+        <InlineCode>small</InlineCode>, <InlineCode>large</InlineCode>, or <InlineCode>default</InlineCode>
+      </>
+    }
```
Comment (colebemis) [inline docs/content/ProgressBar.mdx:41]:
```suggestion
    type={`| 'small'
| 'large'
| 'default'`}
```

File: docs/content/ProgressBar.mdx [hunk 18]
```diff
@@ -20,12 +26,43 @@ If you'd like to use ProgressBar inline, pass the `inline` boolean prop & **be s
...
+    defaultValue="default"
```
Comment (colebemis) [inline docs/content/ProgressBar.mdx:42]:
```suggestion
    defaultValue="'default'"
```

File: docs/content/ProgressBar.mdx [hunk 19]
```diff
@@ -20,12 +26,43 @@ If you'd like to use ProgressBar inline, pass the `inline` boolean prop & **be s
 </>
 ```
 
-## Component props
+## Props
+
+### ProgressBar
+
+<PropsTable>
+  <PropsTableRow name="progress" type="number" description="Used to set the size of the green bar" />
```
Comment (colebemis) [inline docs/content/ProgressBar.mdx:34]: Is this required?

File: docs/content/ProgressBar.mdx [hunk 20]
```diff
@@ -20,12 +26,43 @@ If you'd like to use ProgressBar inline, pass the `inline` boolean prop & **be s
...
+    type={
+      <>
+        <InlineCode>small</InlineCode>, <InlineCode>large</InlineCode>, or <InlineCode>default</InlineCode>
+      </>
+    }
+    defaultValue="default"
+    description="Controls the height of the progress bar. If omitted, height is set to the default height."
+  />
+  <PropsTableRow name="inline" type="boolean" defaultValue="false" description="Styles the progress bar inline" />
+  <PropsTableRow name="bg" type="string" defaultValue="bg.successInverse" description="Set the progress bar color" />
```
Comment (colebemis) [inline docs/content/ProgressBar.mdx:46]:
```suggestion
  <PropsTableRow name="bg" type="string" defaultValue="'bg.successInverse'" description="Set the progress bar color" />
```

File: docs/content/ProgressBar.mdx (hunk 19 above)
Comment (rezrah) [reply docs/content/ProgressBar.mdx:34]: As in a required prop? No, it's optional. Defaults to 0 if it's not passed in.

File: docs/content/Popover.md (hunk 14 above)
Comment (rezrah) [reply docs/content/Popover.md:166]: @colebemis This doesn't seem to work with html tags, getting an empty row. Any ideas? Screenshot 2022-01-21 at 14 16 36

File: docs/content/Popover.md (hunk 12 above)
Comment (rezrah) [reply docs/content/Popover.md:124]: Same comment as before, this suggestion results in an empty row? @colebemis

**Review by colebemis — APPROVED** [review]
🚢

**Pull request merged**